packages/react-heartwood-components/src/components/Forms/components/DurationInput/DurationInput.js

```diff
@@ -45,13 +52,41 @@ export default class DurationInput extends Component<Props, State> {
 		noResultsSubtitle: 'Please adjust your search and try again.'
 	}
 
+	static getDerivedStateFromProps(props, state) {
+		if (props.value !== state.prevPropsValue) {
+			return {
+				prevPropsValue: props.value,
+				value: DurationInput.minutesToStr(props.value)
+			}
+		}
+		return null
+	}
+
+	static minutesToStr = (num: number): string => {
```

Comment:
Note, but output types in flow are usually inferred. you don't need to include them unless you want to force the output to a type (useful sometimes but not always). In this case I think you're fine to omit it.